indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -457,3 +458,78 @@ private void registerWithCloser(Closer closer)
     }
   }
 }
+
+/**
+ * Make an iterable of space delimited strings... unless there are quotes, which it preserves
+ * It will close unbalanced quotes
+ */
+class QuotableWhiteSpaceSplitter implements Iterable<String>
+{
+  private final String string;
+  private static final Splitter quoteSplitter = Splitter.on('"');
+  private static final Splitter spaceSplitter = Splitter.on(CharMatcher.BREAKING_WHITESPACE);
+
+  public QuotableWhiteSpaceSplitter(String string)
+  {
+    this.string = Preconditions.checkNotNull(string);
+  }
+
+  @Override
+  public Iterator<String> iterator()
+  {
+    final Iterable<String> quoted = quoteSplitter.split(string);
+    return new Iterator<String>()
+    {
+      final Iterator<String> quoteIterator = quoted.iterator();
+      Iterator<String> spaceIterator = Iterators.emptyIterator();
+
+      @Override
+      public boolean hasNext()
+      {
+        return spaceIterator.hasNext() || quoteIterator.hasNext();
+      }
+
+      @Override
+      public String next()
+      {
+        if (spaceIterator.hasNext()) {
+          final String prefix = spaceIterator.next();
+          if (spaceIterator.hasNext()) {
+            // No quote
+            return prefix;
+          }
+          if (!quoteIterator.hasNext()) {
+            // End of parse
+            return prefix;
+          }
+          // We have a quoted string. Grab it and advance the space iterator.
+          final String quoted = quoteIterator.next();
+          if (quoteIterator.hasNext()) {
+            spaceIterator = spaceSplitter.split(quoteIterator.next()).iterator();
+          } else {
+            spaceIterator = Iterators.emptyIterator();
+          }
+          return String.format(
+              "%s\"%s\"%s",
+              prefix,
+              quoted,
+              spaceIterator.hasNext() ? spaceIterator.next() : ""
+          );
+        }
+
+        if (!quoteIterator.hasNext()) {
+          // hasNext should have failed
+          throw new IndexOutOfBoundsException();
+        }
+
+        spaceIterator = spaceSplitter.split(quoteIterator.next()).iterator();
+        return next();
+      }
+
+      @Override
+      public void remove() {
+        throw new UnsupportedOperationException("remove");
+      }
+    };
+  }
+}
```

Comment:
came across http://stackoverflow.com/a/19946701 , seems simpler ... see if it makes sense to use the CharMatcher.